core/src/avm1/activation.rs

```diff
@@ -1981,7 +1981,12 @@ impl<'a, 'gc> Activation<'a, 'gc> {
                 return self.set_target(&target);
             }
             Value::Undefined => {
-                self.set_target_clip(None);
+                // In SWF6 and below, SetTarget2 on an undefined object resets the target to the base clip
+                if self.swf_version() > 6 {
```

Comment:
Weird, the SWFs that were fixed by https://github.com/ruffle-rs/ruffle/pull/12956 were both SWFv6. Have you tested those SWFs with this PR?

Reply:
I just did and you're right. Black 2 goes back to locking up.
[308750_BLACK.zip](https://github.com/ruffle-rs/ruffle/files/12701985/308750_BLACK.zip)
Here's the actual game as opposed to the NG launcher.

Reply:
I retract part of my comment this morning. This PR does work against both Curveball and Black 2 as expected. I did not properly clone the branch when I tested earlier today :P 
I advocate for this to be approved and merged.

Reply:
Actually both games are SWFv7 and should not be affected by this PR. @sombraguerrero Please test both games and let us know what you find. This PR should be fine but please confirm.

Reply:
Yes, confirmed. Both Black 1 and 2 are working with the PR and JPEXs reads them both as 7.